### PR TITLE
Fix/recipe host matching

### PR DESCRIPTION
**Pull request by Julien4218**

### CODE REVIEW

**Comment by Julien4218** [comment]
https://newrelic.atlassian.net/browse/VIRTUOSO-1053

File: internal/install/recipes/recipe_repository.go [hunk 1]
```diff
@@ -117,9 +117,16 @@ func findMaxMatch(matches []recipeMatch) recipeMatch {
...
 	for _, match := range matches {
 		if result == nil {
-			result = &match
+			result = &recipeMatch{
+				recipe:     match.recipe,
+				matchCount: match.matchCount,
+			}
 		} else {
 			if match.matchCount > result.matchCount {
+				result = &recipeMatch{
```
Comment (ctrombley) [inline internal/install/recipes/recipe_repository.go:126]: This assignment is unnecessary is we're setting `result` again on line 130, right?

Comment (Julien4218) [reply internal/install/recipes/recipe_repository.go:126]: yep, good catch.

Comment (Julien4218) [reply internal/install/recipes/recipe_repository.go:126]: fixed

**Review by ctrombley — APPROVED** [review]

**Pull request merged**